### PR TITLE
Docker release

### PR DESCRIPTION
**Pull request by justusschock**
# Before submitting

- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to update the docs?   
- [ ] Did you write any new necessary tests?  
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CHANGELOG.md)?

## What does this PR do?
Fixes # (issue).

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: .github/workflows/docker_builds.yml [hunk 1]
```diff
@@ -19,28 +19,26 @@ jobs:
...
       with:
         name: pytorchlightning/pytorch_lightning
         username: ${{ secrets.DOCKER_USERNAME }}
         password: ${{ secrets.DOCKER_PASSWORD }}
         dockerfile: docker/Dockerfile
         buildargs: PYTHON_VERSION=${{ matrix.python_version }},PYTORCH_VERSION=${{ matrix.pytorch_version }},LIGHTNING_VERSION=${{ steps.get_version.outputs.VERSION }}
-        tags: "latest_py${{ matrix.python_version }}_torch${{ matrix.pytorch_version }}"
-  
-
+        tags: "latest-py${{ matrix.python_version }}-torch${{ matrix.pytorch_version }}"
```
Comment (Borda) [inline .github/workflows/docker_builds.yml:44]: what is the latest? like latest from commit from master?

File: .github/workflows/docker_builds.yml [hunk 2]
```diff
@@ -19,28 +19,26 @@ jobs:
       if: contains(github.ref, 'refs/tags/')
       id: get_version
       run: echo ::set-output name=VERSION::${GITHUB_REF/refs\/tags\//}
-    - uses: actions/checkout@2.14
+    - uses: actions/checkout@v2
     - name: Publish Releases to Docker
       # only on releases
-      uses: elgohr/Publish-Docker-Github-Action@master
+      uses: elgohr/Publish-Docker-Github-Action@2.14
       if: contains(github.ref, 'refs/tags/') && !contains(${{ steps.get_version.outputs.VERSION }}, 'rc') %% !contains(${{ steps.get_version.outputs.VERSION }}, 'dev')
```
Comment (Borda) [inline .github/workflows/docker_builds.yml:26]: could it be simplified, if there are only digits and .

File: .github/workflows/docker_builds.yml [hunk 3]
```diff
@@ -19,28 +19,26 @@ jobs:
...
       with:
         name: pytorchlightning/pytorch_lightning
         username: ${{ secrets.DOCKER_USERNAME }}
         password: ${{ secrets.DOCKER_PASSWORD }}
         dockerfile: docker/Dockerfile
         buildargs: PYTHON_VERSION=${{ matrix.python_version }},PYTORCH_VERSION=${{ matrix.pytorch_version }},LIGHTNING_VERSION=${{ steps.get_version.outputs.VERSION }}
-        tags: "${{ steps.get_version.outputs.VERSION }}_py${{ matrix.python_version }}_torch${{ matrix.pytorch_version }},stable_py${{ matrix.python_version }}_torch${{ matrix.pytorch_version }}"
+        tags: "${{ steps.get_version.outputs.VERSION }}-py${{ matrix.python_version }}-torch${{ matrix.pytorch_version }},stable-py${{ matrix.python_version }}-torch${{ matrix.pytorch_version }}"
```
Comment (Borda) [inline .github/workflows/docker_builds.yml:33]: with the stable, it assumes that tag/releases are added chronologically...

File: .github/workflows/docker_builds.yml (hunk 2 above)
Comment (justusschock) [reply .github/workflows/docker_builds.yml:26]: I don't know how far contains supports regex here...

File: .github/workflows/docker_builds.yml (hunk 1 above)
Comment (justusschock) [reply .github/workflows/docker_builds.yml:44]: Yes the latest (unstable, not yet released) version (from master in our case)

File: .github/workflows/docker_builds.yml (hunk 3 above)
Comment (justusschock) [reply .github/workflows/docker_builds.yml:33]: Yes, but I guess we would never add a release after it's successor?

File: .github/workflows/docker_builds.yml (hunk 1 above)
Comment (Borda) [reply .github/workflows/docker_builds.yml:44]: Then I would rather call it nightly, at least my association with latest is stable...

File: .github/workflows/docker_builds.yml (hunk 3 above)
Comment (Borda) [reply .github/workflows/docker_builds.yml:33]: Agree that it an edge case, so just add a note here/comment..

File: .github/workflows/docker_builds.yml (hunk 1 above)
Comment (justusschock) [reply .github/workflows/docker_builds.yml:44]: or simply unstable? Because nightly for me refers to one build every night (which might not be given here)

Comment (Borda) [reply .github/workflows/docker_builds.yml:44]: I would stay with nightly... the nigh is for active repos to reduce nb builds just once overnight in our case we do not have so many commits to master every day so I guess it is the same...
I think if we name it unstable no one will use it :D
but yes, to be consistent we can make a crone for Docker build, right?
https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#onschedule

Comment (Borda) [inline .github/workflows/docker_builds.yml:44]:
```suggestion
        tags: "nightly-py${{ matrix.python_version }}-torch${{ matrix.pytorch_version }}"
```

**Comment by justusschock** [comment]
@williamFalcon this is ready to merge :)

**Pull request merged**